syne_tune/optimizer/schedulers/transfer_learning/zero_shot.py

```diff
@@ -0,0 +1,151 @@
+            sort_transfer_learning_evaluations = False
+        if use_surrogates:
+            sort_transfer_learning_evaluations = False
+            transfer_learning_evaluations = self._create_surrogate_transfer_learning_evaluations(
+                config_space, transfer_learning_evaluations, metric
+            )
+        warning_message = 'This searcher assumes that each hyperparameter configuration occurs in all tasks. '
+        scores = list()
+        hyperparameters = None
+        for task_name, task_data in transfer_learning_evaluations.items():
+            assert hyperparameters is None or task_data.hyperparameters.shape == hyperparameters.shape, warning_message
+            hyperparameters = task_data.hyperparameters
+            if sort_transfer_learning_evaluations:
+                hyperparameters = task_data.hyperparameters.sort_values(list(task_data.hyperparameters.columns))
+            idx = hyperparameters.index.values
+            avg_scores = task_data.objective_values(metric).mean(axis=1)
+            if self._mode == 'max':
+                avg_scores = avg_scores.max(axis=1)[idx]
+            else:
+                avg_scores = avg_scores.min(axis=1)[idx]
+            scores.append(avg_scores)
+        if not use_surrogates:
+            if len(transfer_learning_evaluations) > 1:
+                logger.warning(warning_message + 'If this is not the case, this searcher fails without a warning.')
+            if not sort_transfer_learning_evaluations:
+                hyperparameters = hyperparameters.copy()
+        hyperparameters.reset_index(drop=True, inplace=True)
+        self._hyperparameters = hyperparameters
+        sign = 1 if self._mode == 'min' else -1
+        self._scores = sign * pd.DataFrame(scores)
+        self._ranks = self._update_ranks()
+
+    def _create_surrogate_transfer_learning_evaluations(
+            self, config_space: Dict, transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            metric: str) -> Dict[str, TransferLearningTaskEvaluations]:
+        """
+        Creates transfer_learning_evaluations where each configuration is evaluated on each task using surrogate models.
+        """
+        surrogate_transfer_learning_evaluations = dict()
+        for task_name, task_data in transfer_learning_evaluations.items():
+            estimator = BlackboxSurrogate.make_model_pipeline(
+                configuration_space=config_space,
+                fidelity_space={},
+                model=xgboost.XGBRegressor(),
+            )
+            X_train = task_data.hyperparameters
+            y_train = task_data.objective_values(metric).mean(axis=1)
+            if self._mode == 'max':
+                y_train = y_train.max(axis=1)
+            else:
+                y_train = y_train.min(axis=1)
+            estimator.fit(X_train, y_train)
+
+            num_candidates = 10000 if len(config_space) >= 6 else 5 ** len(config_space)
```

Comment:
Nit comment: I would imagine that it is hard to guess the number of samples required just given the dimension (all HPs could be constant but one) and that a constant number of candidates is easier to reason with.